src/components/Accordion/Accordion.tsx

```diff
@@ -1,30 +1,48 @@
 import classNames from "classnames";
 import PropTypes from "prop-types";
-import React, { useState } from "react";
+import React, { useState, HTMLProps } from "react";
 
 import AccordionSection from "../AccordionSection";
+import type {
+  AccordionHeadings,
+  AccordionSectionProps,
+} from "../AccordionSection";
 
-const generateSections = (sections, setExpanded, expanded, titleElement) =>
+const generateSections = (
+  sections: AccordionSectionProps & { key?: string | number }[],
+  setExpanded,
+  expanded,
+  titleElement
+) =>
   sections.map(({ key, ...props }, i) => (
     <AccordionSection
       expanded={expanded}
       key={key || i}
-      sectionKey={key}
+      sectionKey={key?.toString()}
       setExpanded={setExpanded}
       titleElement={titleElement}
       {...props}
     />
   ));
 
+type Props = {
+  sections: typeof AccordionSection[];
```

Comment:
So I guess this would become `AccordionSectionProps & { key?: string | number }[]`.

I think typeof gives the type of the component, not the props:

<img width="506" alt="Screen Shot 2020-07-28 at 3 29 04 pm" src="https://user-images.githubusercontent.com/361637/88622971-38f7ad00-d0e7-11ea-8e1f-39ba6754bd19.png">